templates/zoom-parts/username-section.tpl

```diff
@@ -14,6 +14,12 @@
     <a id="UsernameUsernameList" class="btn btn-small" target="_blank" href="https://en.wikipedia.org/w/index.php?title=Special%3AListUsers&amp;username={$usernamerawunicode|escape:'url'}&amp;group=&amp;limit=1" onMouseUp="$('#UsernameUsernameList').addClass('btn-visited');">Username list</a>
     <a id="UsernameMainspaceSearch" class="btn btn-small" target="_blank" href="https://en.wikipedia.org/w/index.php?title=Special%3ASearch&amp;profile=advanced&amp;search={$usernamerawunicode|escape:'url'}&amp;fulltext=Search&amp;ns0=1&amp;redirs=1&amp;profile=advanced" onMouseUp="$('#UsernameMainspaceSearch').addClass('btn-visited');">Wikipedia mainspace search</a>
     <a id="UsernameGoogleSearch" class="btn btn-small" target="_blank" href="{$baseurl}/redir.php?tool=google&amp;data={$usernamerawunicode|escape:'url'}" onMouseUp="$('#UsernameGoogleSearch').addClass('btn-visited');">Google search</a>
+    {if $showinfo}
+      <a id="EmailVisit" class="btn btn-small" target="_blank" href="http://{$emailurl}" onMouseUp="$('#EmailVisit').addClass('btn-visited');">Visit email domain</a>
```

Comment:
One approach you could take is use the redirect tool we already use for other Labs tools, and this has the benefit of also removing referrer data.  Pass the URL-encoded domain to the redirect tool, have the redirect tool check the domain, then if it's okay redirect the user.